docs/create-project-shapefiles.md

```diff
@@ -0,0 +1,69 @@
+# Current approach of handling shapefiles
+
+## Custom planning area
+
+Frontend sends Shapefile which is persisted in geo database and its ID and 
+GeoJSON is returned to make it being displayed on map by frontend (not using 
+tiles)
+
+Then, when submitting whole creation, given ID is used.
+
+There is some simple mechanism that collects unused persisted planning areas.
+
+## Custom feature
+
+Frontend sends Shapefile which is persisted in geo database and attached to 
+scenario (without explicit marking it as used). Neither ID nor GeoJSON is 
+returned.
+
+## Protected area
+
+Frontend sends Shapefile which is persisted in geo database and attached to 
+project (without explicit marking it as used). Neither ID nor GeoJSON is
+returned.
+
+## Cost surface
+
+Frontend sends Shapefile which is persisted and processed - values are 
+attached to scenario. Nothing is returned.
+
+# Possible approaches of creating a project
+
+## Full payload
+
+Approach similar to `Custom planning area` should be used whether it is a 
+planning area or grid upload.
+
+Pros:
+- using similar flow for creating, as per custom planning area
+- keeping "everything" as required - clear statement what is needed
+- mostly implemented logic on frontend
+
+Cons:
+- keeping "everything" as required - may be hard to maintain the more steps 
+  come, cannot edit them step by step
+- a need to display custom, returned GeoJSON manually on frontend instead 
+  using map tiles (although already implemented)
+- keeping different approaches of how uploading shapefile works in different 
+  parts of the system
+- requires some further adjustments on backend
+- requires to keep API's to proxy request to Geoprocessing
+
+## Step by step payload
+
```

Comment:
To the question raised by @hotzevzl, about changing the architecture at this stage,  do we know more or less how long this might take? 

And also, could someone explain to me the current approach? 

If we have detected an issue, it might be better to fix it rather than keep adding things on top. 

So my questions would be: 
- How long would this take (the refactor)? 
- Some examples of developments still to come that will benefit from this change. 
- If we face this change, is it only that we are improving the current, or that it will be easier to get subsequentnt features?

Reply:
PROS: 

- It seems to be the most logical way to do it, even though it is not consistent with the rest of the uploads of the application, but I do not see that this is even relevant. Each upload has its needs.

CONS: 

- [ ] Need API and FE adjustments.

- [ ] Need to check with the client if this approach is the one they need.